From 60e21aac0fa45341aac5a1b0e5ca5e89922bba42 Mon Sep 17 00:00:00 2001
From: Sam Mortimer <sam@mortimer.me.uk>
Date: Fri, 9 Nov 2018 17:14:45 -0800
Subject: [PATCH 7/7] LineageParts: Update styles to use internal API

*) lineage-sdk StyleInterface is now internal.

*) Replace lineageos.permission.CHANGE_STYLE with
   AOSP android.permission.CHANGE_OVERLAY_PACKAGES.

Change-Id: Ia62a803a68bf59ae07cc0f444970dc5083b3bb3f
---
 AndroidManifest.xml                                |  2 +-
 .../lineageparts/style/StylePreferences.java       | 14 ++++++++------
 .../lineageparts/style/util/AccentUtils.java       |  4 ++--
 3 files changed, 11 insertions(+), 9 deletions(-)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index fa361fd..dba419a 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -36,9 +36,9 @@
     <uses-permission android:name="android.permission.BIND_DEVICE_ADMIN" />
     <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
     <uses-permission android:name="android.permission.READ_SEARCH_INDEXABLES" />
+    <uses-permission android:name="android.permission.CHANGE_OVERLAY_PACKAGES" />
 
     <uses-permission android:name="lineageos.permission.MANAGE_REMOTE_PREFERENCES" />
-    <uses-permission android:name="lineageos.permission.CHANGE_STYLE" />
 
     <protected-broadcast android:name="lineageos.intent.action.UPDATE_POWER_MENU" />
     <protected-broadcast android:name="lineageos.platform.app.profiles.PROFILES_STATE_CHANGED" />
diff --git a/src/org/lineageos/lineageparts/style/StylePreferences.java b/src/org/lineageos/lineageparts/style/StylePreferences.java
index 7df3fbe..86a9411 100644
--- a/src/org/lineageos/lineageparts/style/StylePreferences.java
+++ b/src/org/lineageos/lineageparts/style/StylePreferences.java
@@ -15,6 +15,8 @@
  */
 package org.lineageos.lineageparts.style;
 
+import static android.Manifest.permission.CHANGE_OVERLAY_PACKAGES;
+
 import android.Manifest;
 import android.app.Activity;
 import android.app.AlertDialog;
@@ -45,17 +47,17 @@ import org.lineageos.lineageparts.style.util.AccentAdapter;
 import org.lineageos.lineageparts.style.util.AccentUtils;
 import org.lineageos.lineageparts.style.util.UIUtils;
 
+import lineageos.providers.LineageSettings;
+import org.lineageos.internal.style.StyleInterface;
+import org.lineageos.internal.style.Suggestion;
+
 import java.util.Arrays;
 import java.util.List;
 
-import lineageos.providers.LineageSettings;
-import lineageos.style.StyleInterface;
-import lineageos.style.Suggestion;
-
 public class StylePreferences extends SettingsPreferenceFragment {
     private static final String TAG = "StylePreferences";
     private static final String CHANGE_STYLE_PERMISSION =
-            lineageos.platform.Manifest.permission.CHANGE_STYLE;
+            CHANGE_OVERLAY_PACKAGES;
     private static final int REQUEST_CHANGE_STYLE = 68;
 
     private Preference mStylePref;
@@ -424,4 +426,4 @@ public class StylePreferences extends SettingsPreferenceFragment {
     private interface Callback {
         void onDone(Suggestion suggestion);
     }
-}
\ No newline at end of file
+}
diff --git a/src/org/lineageos/lineageparts/style/util/AccentUtils.java b/src/org/lineageos/lineageparts/style/util/AccentUtils.java
index 0e47490..06ca2d5 100644
--- a/src/org/lineageos/lineageparts/style/util/AccentUtils.java
+++ b/src/org/lineageos/lineageparts/style/util/AccentUtils.java
@@ -29,7 +29,7 @@ import org.lineageos.lineageparts.R;
 import org.lineageos.lineageparts.style.models.Accent;
 import org.lineageos.lineageparts.style.models.StyleStatus;
 
-import lineageos.style.StyleInterface;
+import org.lineageos.internal.style.StyleInterface;
 
 import java.util.ArrayList;
 import java.util.List;
@@ -105,4 +105,4 @@ public final class AccentUtils {
         StyleStatus accentStatus = accent.getSupportedStatus();
         return accentStatus == StyleStatus.DYNAMIC || currentStatus == accentStatus;
     }
-}
\ No newline at end of file
+}
-- 
2.17.1

